vault/identity_store_oidc_test.go

```diff
@@ -937,12 +937,32 @@ func TestOIDC_PeriodicFunc(t *testing.T) {
 				numPublicKeys int
 			}{
 				{1, 2, 2},
-				{2, 3, 3},
-				{3, 3, 3},
-				{4, 3, 3},
-				{5, 3, 3},
-				{6, 3, 3},
-				{7, 3, 3},
+				{2, 2, 2},
+				{3, 2, 2},
+				{4, 2, 2},
+				{5, 2, 2},
+				{6, 2, 2},
+				{7, 2, 2},
+			},
+		},
+		{
+			// don't set SigningKey to ensure its non-existence can be handled
+			&namedKey{
+				name:            "test-key-nil-signing-key",
+				Algorithm:       "RS256",
+				VerificationTTL: 1 * cyclePeriod,
+				RotationPeriod:  1 * cyclePeriod,
+				KeyRing:         append([]*expireableKey{}, &expireableKey{KeyID: id}),
+				SigningKey:      nil,
+				NextSigningKey:  jwk,
+				NextRotation:    time.Now(),
+			},
+			[]struct {
+				cycle         int
+				numKeys       int
+				numPublicKeys int
+			}{
+				{1, 2, 2},
 			},
```

Comment:
Can we also add a third case where both `SigningKey` and `NextSigningKey` are `nil` (something that can happen with an "old" key, one without `NextSigningKey`, that got corrupted by a 1.9.0 upgrade)?

On a similar note, a case. were all expected values are present would also be good to test.

Reply:
Will do, thanks